src/Aspire.Hosting/ResourceBuilderExtensions.cs

```diff
@@ -1660,6 +1672,99 @@ public static IResourceBuilder<T> WithRelationship<T>(
         return builder.WithAnnotation(new ResourceRelationshipAnnotation(resource, type));
     }
 
+    /// <summary>
+    /// Adds a <see cref="ResourceRelationshipAnnotation"/> to the resource annotations to add a reference to another resource.
+    /// </summary>
+    /// <typeparam name="T">The type of the resource.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="resource">The resource that the relationship is to.</param>
+    /// <returns>A resource builder.</returns>
+    public static IResourceBuilder<T> WithResourceRelationship<T>(
```

Comment:
What is the difference between this and `WithRelationship`? Just hardcoding a relationship type of `Reference`?

`ResourceRelationship` is a generic name that doesn't mean anything. Aren't all relationships between resources resource relationships?

Shouldn't this be called `WithReferenceRelationship`?

Reply:
I'm not hung up on the naming, and yes it's to avoid the magic "Reference" string.